### ISSUE TITLE
Memory Leak in version 2.0.0

### ISSUE THREAD

**Issue by angelnm** [mjpost/sacrebleu#167]
Hi,

In my Python program, I call multiple times to the function bleu.corpus_score, and I have detected that the memory that the program needs increases with time. 

I have isolated the problem to the line of code where I call the corpus_score function.

I have changed my sacrebleu version from 2.0.0 to 1.5.1 and the memory leak disappears. There is a problem with the new version that does not free memory correctly.

Really appreciate your help!

**Comment by angelnm** [comment]
Yes, of course

In this code snippet I'm studying the evolution of the BLEU score while adding incrementally the words from the hypothesis. That's the purpose of the loop inside the code.
```Python

bleu = BLEU()
def bleu_reward(c_src, c_hyp):
    last = 0
    c_rew= []

    hypothesis = ['<null>']*len(c_hyp)
    for idx, word in enumerate(c_hyp):
        hypothesis[idx] = word
        score= bleu.corpus_score([' '.join(hypothesis)], [[' '.join(c_src)]]).score
        c_rew.append(score - last)
        last = score

    return c_rew
```
I call this functions for every pair of sentences hypothesis/reference

**Comment by mjpost** [comment]
I wonder if we should turn off the caching by default.

**Comment by ozancaglayan** [comment]
I just tried running the above code in a tight loop, but using always the same sentence pair and the memory usage was not increasing. That's why I made the assumption that it's probably the caching that increases the memory usage as @martinpopel suggested. I also agree that the benefits probably outweighs the memory increase, but let's here from @angelnm again their usage pattern. I wonder whether they observe something more aggressive/prohibitive in their case.
